sdk/nodejs/provider/experimental/ComponentProvider.md

```diff
@@ -0,0 +1,267 @@
+# Component Provider
+
+The `@pulumi/pulumi/provider/experimental` package allows writing a Pulumi provider in TypeScript with minimal boilerplate, letting pulumi infer the schema. The provider can be published to Git or referenced locally.
+
+## Building a Component Provider
+
+Create a new directory that will hold the provider implementation.
+
+First we need to let Pulumi's plugin system know which language runtime should be used to run the provider plugin. Create a `PulumiPlugin.yaml` file with the contents:
+
+```yaml
+runtime: nodejs
+```
+
+We also need to create a `package.json` file to define the project and list the dependencies of the provider.  `@pulumi/pulumi` needs to be at least version 3.153.0.
+
+```txt
+{
+    "name": "main",
+    "main": "index.ts",
+    "devDependencies": {
+        "@types/node": "^18",
+        "typescript": "^5.0.0"
+    },
+    "dependencies": {
+        "@pulumi/pulumi": "^3.153.0",
+		"@pulumi/random": "^4.0.0"
+    }
+}
+```
+
+This uses a prerelease version of the NodeJS Pulumi SDK that includes the experimental component provider feature. We'll use the `@pulumi/random` package to generate a random greeting.
+
+The resources provided by the provider are are defined as TypeScript classes that subclass `pulumi.ComponentResource`. Create a new file `myComponent.ts` to hold the component implementation:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+
+export interface MyComponentArgs {
+}
+
+export class MyComponent extends pulumi.ComponentResource {
+    constructor(name: string, args: MyComponentArgs, opts?: pulumi.ComponentResourceOptions) {
+	super("provider:index:MyComponent", name, args, opts);
+    }
+}
+```
+
+> [!NOTE]
+> The type string `my-provider:index:MyComponent` follows the pattern `<package>:<module>:<type>`. The package is your provider's name, module must be `index`. The `type` must match the class name.
+
+Next, create a `index.ts` file to host the provider:
+
+```typescript
+import * as experimental from "@pulumi/pulumi/provider/experimental";
+
+experimental.componentProviderHost();
```

Comment:
I’d avoid importing as `experimental` alias, just import the function itself, easier to change later